x-pack/plugins/reporting/server/usage/decorate_range_stats.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CSV_JOB_TYPE, PDF_JOB_TYPE, PNG_JOB_TYPE } from '../../common/constants';
+import { AvailableTotal, FeatureAvailabilityMap, RangeStats, ReportingFeature } from './';
+
+function getForFeature(
+  range: Partial<RangeStats>,
+  featureKey: ReportingFeature,
+  featureAvailability: FeatureAvailabilityMap,
+  additional?: any
+): AvailableTotal & typeof additional {
+  const isAvailable = (feature: ReportingFeature) =>
+    featureAvailability[feature] != null ? featureAvailability[feature] : false;
+  const jobType = range[featureKey] || { total: 0, ...additional };
+
+  // merge the additional stats for the jobType
+  type AdditionalType = { [K in keyof typeof additional]: K };
+  const filledAdditional: AdditionalType = {};
+  if (additional) {
+    Object.keys(additional).forEach(k => {
+      filledAdditional[k] = { ...additional[k], ...jobType[k] };
+    });
+  }
+
+  return {
+    available: isAvailable(featureKey),
+    total: jobType.total,
+    ...filledAdditional,
+  };
+}
+
+/*
+ * Decorate range stats (stats for last day, last 7 days, etc) with feature
+ * availability booleans, and zero-filling for unused features
+ */
+export const decorateRangeStats = (
+  range: Partial<RangeStats> = {},
+  featureAvailability: FeatureAvailabilityMap
+): RangeStats => {
+  return {
+    _all: range._all || 0,
+    status: { completed: 0, failed: 0, ...range.status },
+    [CSV_JOB_TYPE]: getForFeature(range, CSV_JOB_TYPE, featureAvailability),
+    [PNG_JOB_TYPE]: getForFeature(range, PNG_JOB_TYPE, featureAvailability),
+    [PDF_JOB_TYPE]: getForFeature(range, PDF_JOB_TYPE, featureAvailability, {
```

Comment:
I'm unsatisfied with building the return object this way, since it hardcodes for known job types.

It makes sense to hardcode the known `printable_pdf` type, since that has extra fields that are handled here. For the other ones, I think it needs a more agnostic method.

Reply:
resolved by https://github.com/elastic/kibana/pull/34327/commits/d159aa31c51e71271d96c500341a7ff171d0cd96